base/test.jl

```diff
@@ -487,7 +491,13 @@ macro testset(args...)
     quote
         $ts = DefaultTestSet($desc)
         add_testset($ts)
-        $(esc(tests))
+        try
+            $(esc(tests))
+        catch err
+            # something in the test block threw an error. Count that as an
+            # error in this test set
+            record($ts, Error(:test_error, :(), err, catch_backtrace()))
```

Comment:
That way instead of the empty-expression `if` above, you could have a separate message that explains clearly that this was a "floating" error instead of an error in a test.